pkg/k8s_utils/secret.go

```diff
@@ -0,0 +1,92 @@
+/**
+ * Copyright 2022 IBM Corp.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+// Package k8s_utils ...
+package k8s_utils
+
+import (
+	"context"
+	b64 "encoding/base64"
+	"fmt"
+	"io/ioutil"
+
+	"github.com/IBM/secret-utils-lib/pkg/utils"
+	"go.uber.org/zap"
+	v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/rest"
+)
+
+// GetSecretData ...
+func GetSecretData(secretname, dataname string, logger *zap.Logger) (string, error) {
+	logger.Info("Fetching secret", zap.String("Secret name", secretname))
+
+	// Fetching cluster config used to create k8s client
+	k8sConfig, err := rest.InClusterConfig()
+	if err != nil {
+		logger.Error("Error fetching in cluster config", zap.Error(err))
+		return "", utils.Error{Description: fmt.Sprintf("Error fetching secret %s - unable to fetch cluster config", secretname), BackendError: err.Error()}
+	}
+
+	// Creating k8s client used to read secret
+	clientset, err := kubernetes.NewForConfig(k8sConfig)
+	if err != nil {
+		logger.Error("Error creating k8s client", zap.Error(err))
+		return "", utils.Error{Description: fmt.Sprintf("Error fetching secret %s - unable to create k8s client", secretname), BackendError: err.Error()}
+	}
+
+	// Reading the namespace in which the pod is deployed
+	byteData, err := ioutil.ReadFile("/var/run/secrets/kubernetes.io/serviceaccount/namespace")
```

Comment:
why this is hardcoded ?

Reply:
As discussed, this will be used. the path will be moved to constants file